protostar/commands/format/format_command.py

```diff
@@ -0,0 +1,147 @@
+from logging import Logger
+from typing import List, Optional, Tuple
+from pathlib import Path
+
+from starkware.cairo.lang.compiler.parser import parse_file
+from starkware.cairo.lang.compiler.parser_transformer import ParserError
+from starkware.cairo.lang.compiler.ast.formatting_utils import FormattingError
+
+from protostar.cli import Command
+from protostar.protostar_exception import ProtostarExceptionSilent
+from protostar.utils import log_color_provider
+
+from protostar.commands.format.formatting_summary import FormatingSummary
+from protostar.commands.format.formatting_result import (
+    BrokenFormattingResult,
+    CorrectFormattingResult,
+    IncorrectFormattingResult,
+)
+
+
+class FormatCommand(Command):
+    def __init__(self, project_root_path: Path, logger: Logger) -> None:
+        super().__init__()
+        self._project_root_path = project_root_path
+        self._logger = logger
+
+    @property
+    def example(self) -> Optional[str]:
+        return "$ protostar format"
+
+    @property
+    def name(self) -> str:
+        return "format"
+
+    @property
+    def description(self) -> str:
+        return "Format cairo sourcecode."
+
+    @property
+    def arguments(self) -> List[Command.Argument]:
+        return [
+            Command.Argument(
+                name="target",
+                description=("Target to format, can be a file or a directory."),
+                type="str",
+                is_array=True,
+                is_positional=True,
+                default=["."],
+            ),
+            Command.Argument(
+                name="check",
+                description=(
+                    "Run in 'check' mode. Exits with 0 if input is formatted correctly."
+                    "Exits with 1 if formatting is required."
+                ),
+                type="bool",
+                is_required=False,
+                default=False,
+                short_name="c",
+            ),
+            Command.Argument(
+                name="log-formatted",
+                description=("Log information about already formatted files."),
+                type="bool",
+                is_required=False,
+                default=False,
+            ),
+            Command.Argument(
+                name="ignore-broken",
+                description=("Ignore broken files."),
+                type="bool",
+                is_required=False,
+                default=False,
+            ),
+        ]
+
+    async def run(self, args):
+        try:
+            summary, any_unformatted_or_broken = self.format(
+                args.target, args.check, args.log_formatted, args.ignore_broken
+            )
+        except BaseException as exc:
+            self._logger.fatal("Command failed.")
+            raise exc
+        summary.log_summary(log_color_provider)
```

Comment:
FYI: There was a discussion on the channel, and we agreed to disable `log_color_proivded` by default and use it as a global variable. (I've just merged the PR that disables color_provider by default.)